### ISSUE TITLE
Customise page has no defaults selected

### ISSUE THREAD

**Issue by FlicHollis** [google/site-kit-wp#5757]
When I first land on the customise TwG page, only 'Type' and 'Post type' options are selected as a default (prominence and colour are not selected by default) so it isn't overly obvious to users as to why the 'Configure Thank with Google' button is greyed out. (There's no text indication to explain to the user they need to select a 'Prominence' and 'Colour' in order to configure).
Could be confusing to have 2 of the selections automatically selected, and not the other 2?

<img width="250" alt="Customise page" src="https://user-images.githubusercontent.com/93320881/186770541-2e6b6262-5867-4bf0-9a2f-d4f7db168a16.png">

This is the same if the user switches to 'Fixed' type.

The 'Position' and 'Colour' settings are not selected by default

Need to check which should be default colour.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by eugene-manuilov** [comment]
> * In `CTAPlacement.js`, set `defaultType` to be `TYPE_OVERLAY` if `getPlacementType( ctaPlacement )` returns `null`, i.e. `const defaultType = getPlacementType( ctaPlacement ) || TYPE_OVERLAY;`
> * For the below components, simply setting a default value does not set them in the store which is required to enable the Configure CTA.
> * In `ProminenceRadio.js`, within the `onChange` callback, set `placement` to be `CTA_PLACEMENT_DYNAMIC_LOW` by default. Move the `onChange()` method to above the `ctaPlacement` definition and call this function if the `getCTAPlacement()` selector is falsy/null, i.e. `const ctaPlacement = useSelect(...) || onChange();`
> * In `ColorRadio.js`, similar to the above, move the `onChange` method  and call it if the `getColorTheme()` selector is falsy/null, i.e. `const currentColor = useSelect( ... ) || onChange();`

@jimmymadon, we don't need to change anything in the js, what we need to do is to add correct defaults to the settings class: https://github.com/google/site-kit-wp/blob/f4972bfccc439edd6efdc46fbb9fb9f200c0bb69/includes/Modules/Thank_With_Google/Settings.php#L64-L66

**Comment by jimmymadon** [comment]
@eugene-manuilov Completely missed that. Thanks - have updated the IB.

**Comment by eugene-manuilov** [comment]
IB ✔️

**Comment by aaemnnosttv** [comment]
@eugene-manuilov while it makes sense to apply the defaults in the settings class, won't this prevent the customization step from appearing during setup since it will be considered connected as soon as the publication ID is set? That's the reason for this point of the AC at least:

> * These defaults should only take effect during the module setup/customization. Only selected/saved values should be considered to determine the module's connected status

**Comment by eugene-manuilov** [comment]
Good call, @aaemnnosttv. I forgot about it. 🤦 Looks like we need to get back to the previous IB defined by Jimmy. I'll update IB.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv IB is updated. Do you mind reviewing it?

**Comment by aaemnnosttv** [comment]
@eugene-manuilov @jimmymadon I'm not sure if `useMount` would be the right approach or not, it might be better to use an effect that only runs once the TwG settings are loaded to ensure we only set values where none are set yet. Of course none should actually be set at this point but it seems like a reasonable precaution to take to ensure they're only applied as defaults. It shouldn't really make a difference because the settings would be preloaded, even if the user refreshed the page on that step.

So an alternative could be to create an action that both waits for settings to be loaded (i.e. `resolveSelect`) before applying the defaults. Then that action could be safely invoked on mount. That might be a bit easier to test as well. What do you think?

**Comment by eugene-manuilov** [comment]
@aaemnnosttv I think an additional action that waits for a selector to be resolved is an overkill for this component. I have updated IB to use the `useEffect` hook, please, take a look. If you still think that a new action is a better approach, let me know and I'll update IB.

**Comment by aaemnnosttv** [comment]
Thanks @eugene-manuilov, that sounds good to me. Just one thought here.

> * Use the aforementioned settings as dependencies for the `useEffect` hook.

This would cause the effect to run every time the settings are changed, which would be more often than we want. As an effect, I think we could use a value from `hasFinishedResolution` as the "main" dependency, and in order to avoid including the actual settings as a dependency, we could use the relevant settings selector functions in the effect instead which wouldn't change.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv IB is updated. I used slightly different approach to determine whether we need to set defaults. Let me know if it works now.

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by asvinb** [comment]
@eugene-manuilov A few things:

- I've noticed that the Post Types value is already set to `[posts]` by default, so i didn't implement it.
- I had to make a few changes to the components so that they rerender when the cta placement changes.

I assigned it to you for CR since you reviewed the IB and I saw you worked on the files recently as well.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- The TwG customization setup step has default values selected for all of the settings: 
- The default values are:
  - Type: Overlay
  - Prominence: Low
  - Color: Blue
  - Post Types: Posts

- The defaults  take effect during the module setup/customization. 
- Ran through a number of tests where I saved the default and also made changes settings. 
- Checked that the correct settings appear when viewing and editing the settings. 

https://user-images.githubusercontent.com/73545194/190649309-58b5f958-1f31-42a3-8038-751311e68d04.mp4
